core/src/test/java/io/questdb/griffin/AlterTableSetTypeRestartTest.java

```diff
@@ -0,0 +1,384 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin;
+
+import io.questdb.AbstractBootstrapTest;
+import io.questdb.Bootstrap;
+import io.questdb.ServerMain;
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.TableToken;
+import io.questdb.cairo.security.AllowAllCairoSecurityContext;
+import io.questdb.cairo.wal.ApplyWal2TableJob;
+import io.questdb.cairo.wal.CheckWalTransactionsJob;
+import io.questdb.cairo.wal.WalUtils;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.Chars;
+import io.questdb.std.Files;
+import io.questdb.std.Misc;
+import io.questdb.std.str.Path;
+import io.questdb.test.tools.TestUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.postgresql.util.PSQLException;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import static io.questdb.griffin.AlterTableSetTypeTest.NON_WAL;
+import static io.questdb.griffin.AlterTableSetTypeTest.WAL;
+import static org.junit.Assert.*;
+
+public class AlterTableSetTypeRestartTest extends AbstractBootstrapTest {
+    private static final Log LOG = LogFactory.getLog(AlterTableSetTypeRestartTest.class);
+
+    private final String tableName = "testtable";
+
+    @BeforeClass
+    public static void setUpStatic() throws Exception {
+        AbstractBootstrapTest.setUpStatic();
+        try {
+            createDummyConfiguration("cairo.wal.supported=true");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testNonWalToWalWithDropTable() throws Exception {
+        TestUtils.assertMemoryLeak(() -> {
+            try (final ServerMain questdb = new ServerMain("-d", root.toString(), Bootstrap.SWITCH_USE_DEFAULT_LOG_FACTORY_CONFIGURATION)) {
+                questdb.start();
+                createTable(tableName, "BYPASS WAL");
+                insertInto(tableName);
+
+                final CairoEngine engine = questdb.getCairoEngine();
+                final TableToken token = engine.getTableToken(tableName);
+
+                // non-WAL table
+                assertFalse(engine.isWalTable(token));
+                assertNumOfRows(engine, tableName, 1);
+                assertConvertFileDoesNotExist(engine, tableName);
```

Comment:
you already have the token, pass ot to `assertConvertFileDoesNotExist`

Reply:
and `assertConvertFileExists`

Reply:
fixed in [509c5d5](https://github.com/questdb/questdb/pull/2947/commits/509c5d5677a0aa8f5bac45ee8878683fc16321fe)